pandas/tests/io/test_parquet.py

```diff
@@ -251,7 +262,7 @@ def test_columns_dtypes(self, engine):
 
         # unicode
         df.columns = [u'foo', u'bar']
-        self.check_round_trip(df, engine, write_kwargs={'compression': None})
```

Comment:
is there a reason you are changing all of this?

Reply:
Well yes, I've took the opportunity to do small refactoring in test_parquet.py.

test_parquet.py `Base.check_round_trip` had some code that was both repeating twice and that I needed to test my new functionality of doing round_trip when `s3` is the destination. So refactored it into `do_round_trip`.

Now i'm using `do_round_trip` in both the new units add to test writing/reading parquet files from s3 and in the rest of the unit testing code that continue ti call the existing `check_round_trip` which uses local file system as it's default destination when calling `do_round_trip`.

So now we have `check_round_trip` that calls `do_round_trip` passing it an `engine_impl` and a local file system `path` and my new unit tests under `TestParquetPyArrow` calling  `do_round_trip`  with a `pyarrow engine impl` and `'s3://pandas-test/pyarrow.parquet'` and `TestParquetFastParquet` which calls  `do_round_trip` passing it a `fastparquet engine impl` and an ` 's3://pandas-test/fastparquet.parquet'` (but expecting a `NotImplementedError` exception).

As for `write_kwargs = {'compression': None}` - It was repeating in some of the unit tests in the class. So as part of the refactoring I took the opportunity to set it as a default, if write_kwargs is `None`.